Mage.Sets/src/mage/cards/w/WalkingBulwark.java

```diff
@@ -92,7 +92,7 @@ public boolean apply(Game game, Ability source) {
         game.addEffect(new CanAttackAsThoughItDidntHaveDefenderTargetEffect(Duration.EndOfTurn), source);
         FilterCreaturePermanent filter = new FilterCreaturePermanent();
         filter.add(new MageObjectReferencePredicate(permanent, game));
-        game.addEffect(new CombatDamageByToughnessEffect(filter, false), source);
+        game.addEffect(new CombatDamageByToughnessEffect(filter, false, Duration.EndOfTurn), source);
```

Comment:
you can simplify this with `.setDuration(Duration.EndOfTurn)`

Reply:
how would i implement a 2nd constructor with duration without calling the super? Sorry kinda rusty with java

Reply:
you make a constructor that takes duration and calls the super and then another one which calls `this()` with the default duration choice

Reply:
a good example would be `BoostTargetEffect`